File: 0338-counting-bits/0338-counting-bits.js
/**
 * @param {number} n
 * @return {number[]}
 */

const countBits = n => {
    let arr = new Array(n + 1);
    arr[0] = 0;
    for (let i = 1; i < arr.length; i++) {
        if (i % 2 === 0) arr[i] = arr[Math.floor(i / 2)];
        else arr[i] = arr[Math.floor(i / 2)] + 1;
    }
    return arr;
};

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        arr = []
        def bitCount(n):
            count = 0
            while n > 0:
                count += (n & 1)
                n >>= 1
            return count
        for i in range(n + 1):
            arr.append(bitCount(i))
        return arr